### PR TITLE
Annotate System.Diagnostics.Process APIs throwing PNSE on iOS/tvOS

### PR DESCRIPTION
**Pull request by MaximLipnin**
This pull request is to annotate `System.Diagnostics.Process` APIs throwing PNSE on iOS/tvOS as unsupported.

https://github.com/dotnet/runtime/issues/47910

### CODE REVIEW

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs [hunk 1]
```diff
@@ -1197,8 +1198,15 @@ private void SetProcessId(int processId)
         ///       component.
         ///    </para>
         /// </devdoc>
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
         public bool Start()
         {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("IOS")) || RuntimeInformation.IsOSPlatform(OSPlatform.Create("TVOS")))
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs:1205]: Can we do this in https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L353 instead?  Also, can it use OperatingSystem.IsIOS / OperatingSystem.IsTvOS?

Comment (MaximLipnin) [reply src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs]: Done, thanks

File: eng/versioning.targets [hunk 2]
```diff
@@ -36,11 +36,20 @@
...
 
+  <!-- Enables warnings for iOS targeted builds -->
+  <ItemGroup Condition="'$(TargetsiOS)' == 'true' and '$(IsTestProject)' != 'true'">
+    <SupportedPlatform Include="ios"/>
+  </ItemGroup>
+  <!-- Enables warnings for tvOS targeted builds -->
+  <ItemGroup Condition="'$(TargetstvOS)' == 'true' and '$(IsTestProject)' != 'true'">
+    <SupportedPlatform Include="tvos"/>
+  </ItemGroup>
+
```
Comment (marek-safar) [inline eng/versioning.targets:52]: Why is this needed ?

/cc @buyaa-n

File: src/libraries/System.Diagnostics.Process/ref/System.Diagnostics.Process.cs [hunk 3]
```diff
@@ -113,16 +113,30 @@ public partial class Process : System.ComponentModel.Component, System.IDisposab
...
         public static System.Diagnostics.Process Start(string fileName) { throw null; }
+        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("ios")]
+        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("tvos")]
         public static System.Diagnostics.Process Start(string fileName, string arguments) { throw null; }
+        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("ios")]
+        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("tvos")]
         public static System.Diagnostics.Process Start(string fileName, System.Collections.Generic.IEnumerable<string> arguments) { throw null; }
         [System.CLSCompliantAttribute(false)]
         [System.Runtime.Versioning.SupportedOSPlatformAttribute("windows")]
+        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("ios")]
```
Comment (marek-safar) [inline src/libraries/System.Diagnostics.Process/ref/System.Diagnostics.Process.cs:133]: It's not needed as it alread has `[System.Runtime.Versioning.SupportedOSPlatformAttribute("windows")]`

File: eng/versioning.targets (hunk 2 above)
Comment (MaximLipnin) [reply eng/versioning.targets:52]: @buyaa-n Thanks, what about `tvOS`? Does it still need to be enabled here?

Comment (buyaa-n) [reply eng/versioning.targets:52]: Yes, for unsupported APIs the analyzer shows warnings only for the platforms included in the [MSBuild SupportedPlatform list](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props#L16-L20), `tvOS` is not there so to see unsupported warning for `tvOS` you can add it to the list like this

Comment (MaximLipnin) [reply eng/versioning.targets]: Removed iOS case

Comment (marek-safar) [reply eng/versioning.targets]: @buyaa-n why is tvOS missing it that list?

Comment (terrajobst) [reply eng/versioning.targets]: The intent of the default list is to model what we think most library authors will care about. We didn't add `Browser` because we believe most libraries don't need to run on WebAssembly nor that they can, due to the restrictions of the sandbox.

Whether or not tvOS or iOS are in that group is for the Xamarin team to decide. But the litmus test will be how much code it would flag. If a large chunk of fairly widely used APIs won't work on iOS or tvOS, we probably shouldn't add either to the default set.

File: src/libraries/System.Security.Cryptography.X509Certificates/src/System.Security.Cryptography.X509Certificates.csproj [hunk 4]
```diff
@@ -14,6 +14,7 @@
          TODO: [AndroidCrypto] Remove ANDROID_OPENSSL_AAR check once Android native crypto is complete -->
     <UseAndroidCrypto Condition="'$(TargetsAndroid)' == 'true' and '$(ANDROID_OPENSSL_AAR)' == ''">true</UseAndroidCrypto>
     <UseAppleCrypto Condition="'$(TargetsOSX)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">true</UseAppleCrypto>
+    <ExcludeAppleCryptoOnMobile Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">true</ExcludeAppleCryptoOnMobile>
```
Comment (danmoseley) [inline src/libraries/System.Security.Cryptography.X509Certificates/src/System.Security.Cryptography.X509Certificates.csproj:17]: I highly recommend not using "negative named" properties. It's often caused confusion in the past because when you set it to false, there's a double negative.

Consider `<IncludeAppleCryptoOnMobile>`

Comment (MaximLipnin) [reply src/libraries/System.Security.Cryptography.X509Certificates/src/System.Security.Cryptography.X509Certificates.csproj]: Addressed, thanks.

**Comment by marek-safar** [comment]
@MaximLipnin I think it'd be easier for the review to land each library annotations as separate PR

**Comment by MaximLipnin** [comment]
> @MaximLipnin I think it'd be easier for the review to land each library annotations as separate PR

Yeah, thanks, I'll limit this pull request to System.Diagnostics.Process related API only.

**Review by rolfbjarne — APPROVED** [review]

**Review by marek-safar — APPROVED** [review]
